### DISCUSSION TITLE
Airflow Backend Database RDS PostgreSQL high CPU usage

### DISCUSSION THREAD

**Discussion by sebinxavi** [apache/airflow#40994] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### If "Other Airflow 2 version" selected, which one?

2.8.4

### What happened?

Hello,

We have noticed 55% CPU usage in the newly deployed RDS Aurora PostgreSQL .

We have connected two airflow environments into this RDS instance.
There are no airflow dags running , but still it is taking 55% usage. Is this normal?

<img width="1035" alt="image" src="https://github.com/user-attachments/assets/716afffe-674f-4031-a1cc-c64605750f02">

The result of performance insight is provided below,

<img width="1558" alt="Screenshot 2024-07-24 at 12 55 50" src="https://github.com/user-attachments/assets/ec39e18f-4b79-446c-a5da-b3d049a17b50">

<img width="396" alt="Screenshot 2024-07-24 at 12 56 06" src="https://github.com/user-attachments/assets/32880341-9607-4200-8d55-0acfd2015e2d">

### What you think should happen instead?

Since no DAGs are running and no DAGs are scheduled, the RDS CPU usage should be very low?

### How to reproduce

Deploy airflow 2.8.4 with airflow official helm chart.
Connect the airflow to external database RDS Aurora PostgreSQL.

### Operating System

EKS - 1.28

### Versions of Apache Airflow Providers

```
$ pip freeze | grep apache-airflow-providers
apache-airflow-providers-amazon==8.19.0
apache-airflow-providers-celery==3.6.1
apache-airflow-providers-cncf-kubernetes==8.0.1
apache-airflow-providers-common-io==1.3.0
apache-airflow-providers-common-sql==1.11.1
apache-airflow-providers-docker==3.9.2
apache-airflow-providers-elasticsearch==5.3.3
apache-airflow-providers-ftp==3.7.0
apache-airflow-providers-google==10.16.0
apache-airflow-providers-grpc==3.4.1
apache-airflow-providers-hashicorp==3.6.4
apache-airflow-providers-http==4.10.0
apache-airflow-providers-imap==3.5.0
apache-airflow-providers-microsoft-azure==9.0.1
apache-airflow-providers-microsoft-mssql==3.4.0
apache-airflow-providers-mysql==5.5.4
apache-airflow-providers-odbc==4.4.1
apache-airflow-providers-openlineage==1.6.0
apache-airflow-providers-postgres==5.10.2
apache-airflow-providers-redis==3.6.0
apache-airflow-providers-sendgrid==3.4.0
apache-airflow-providers-sftp==4.9.0
apache-airflow-providers-slack==8.6.1
apache-airflow-providers-smtp==1.6.1
apache-airflow-providers-snowflake==5.3.1
apache-airflow-providers-sqlite==3.7.1
apache-airflow-providers-ssh==3.10.1
```

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

Airflow is deployed with helm chart
https://github.com/airflow-helm/charts/blob/airflow-8.9.0/charts/airflow/values.yaml

### Anything else?

This occurs every time. 
We are seeing below logs in airflow-scheduler pod. There are no DAGs running and no DAGs scheduled to run.
```
{scheduler_job_runner.py:1803} INFO - Orphaning unreferenced dataset 
{scheduler_job_runner.py:1803} INFO - Orphaning unreferenced dataset 
{scheduler_job_runner.py:1803} INFO - Orphaning unreferenced dataset 
[2024-07-24T09:05:27.309+0000] {kubernetes_executor.py:239} INFO - Found 0 queued task instances
```

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
Converted into discussion - some DB traffic is normal - but some of the qureis indicate that your DAG are written against best practices https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html#top-level-python-code and are using Variables - Airflow continuously parses the DAG files and if top level code has a DB access, it will be happening all the time. Also the query you screenshoted is unrelated to airflow- you should look in your DB what generates it.

Also if you have question and you are not sure if this is an airflow issue - start a discussion not create an issue this migh help you to investigate your problem before you assume it is an Airflow issue.

**Comment by omkar-foss** [comment]
Hey @sebinxavi, I tried out running Airflow idle with no DAGs on Postgres and was unable to see such load. So it doesn't look like the load in your case would be originating from Airflow.

The query associated with `Orphaning unreferenced dataset` in the Airflow scheduler isn't in your queries screenshot, so doesn't look like that's related to your Postgres load either.

But the query you've attached above (`SELECT t.oid, typarray FROM pg_type ...`), I've seen it before with Django. [This Django ticket](https://code.djangoproject.com/ticket/28334) might be useful to you.

If you've a Django (or similar) app pointing to the same Postgres instance as your Airflow, I'd suggest to check that, as chances are that this query (and the load) could be originating from there.
